src/assets/htmx/htmx.md

```diff
@@ -0,0 +1,367 @@
+
+First let's make clicking on the todo item go to a different page that will contain a form for updating the todo item list.
+For that, we need to get an anchor tag instead of a div with the todo text inside a div.
+
+
+On the server side, I will get the task and return the proper form for updating that task.
+
+```go
+func main() {
+    ...
+    http.HandleFunc("/task/", individualTask)
+    ...
+}
+```
+
+Example response from server:
+
+```htmx
+  <div hx-boost="true">
+      <a href="/task/1">clean room</a>
+  </div>
+```
+
+Here the `hx-boost` attribute will replace the body element and add the previous page to the history entry.
+That means the back button on browsers will work perfectly.
+
+![demo](/src/assets/htmx/editpage.gif)
+
+For the actual editing, it will be similar to the other examples.
+
+```HTML
+<div>
+  <div>clean room</div>
+  <form id="myForm2" hx-put="/edit" hx-target="#app">
+    <input type="text" name="task" required="" />
+    <input type="hidden" name="old" value="1" />
+    <input type="submit" value="edit" />
+  </form>
+</div>
+```
+
+Now let's perform deletion on the list.
+
+First, we have to make a delete button.
+
+For that, I will update the list template and add a delete form:
+
+```HTML
+<div hx-boost="true" class="new-element">
+  <a href="/task/1">clean room</a>
+  <form
+    hx-trigger="click"
+    hx-delete="/delete"
+    hx-target="closest .new-element"
+    hx-swap="outerHTML swap:1s"
+  >
+    <input type="hidden" name="task" value="1" />
+    delete
+  </form>
+</div>
+```
+
+Most of the snippets in the delete form are similar to the other examples. But in the `hx-swap="outerHTML swap:1s"` attribute I added an extra value. This value will delay the swapping of the element by 1 second because I want to do a CSS animation during that time.
+
+While htmx is doing the swapping of the element it also adds the `htmx-swapping` class to the element.
+
+Now I can add any CSS effect I want to the `htmx-swapping` class.
+
+```CSS
+.htmx-swapping {
+  opacity: 0;
+  transition: opacity 1s ease-out;
+}
+```
+
+![demo](/src/assets/htmx/delete.gif)
+
+## WebSockets
+
+Currently, if we add a todo task to our list, it won't be shown in another tab or browser opening the same page. This can be achieved using websockets. Fortunately htmx supports websockets.
+
+```HTML
+<div id="ws" hx-ws="connect:/ws" hx-trigger="task"></div>
+```
+
+The `hx-ws` attribute takes the websocket endpoint and makes a connection to it.
+
+Now in my code I just have to create a websocket that listines for input by the user and broadcast it to all clients.
+
+```go
+import(
+    ...
+    "golang.org/x/net/websocket"
+    "sync"
+    ...
+)
+
+func main(){
+    ...
+    server := NewServer()
+    http.Handle("/ws", websocket.Handler(server.websocketHandler)) // handel the saving and broadcasting of data
+    ...
+}
+```
+
+Update the form
+
+```HTML
+<form
+  id="myForm1"
```

Comment:
something more descriptive as id might help the user to understand this example better